leo_fw/src/leo_fw/test_hw.py

```diff
@@ -0,0 +1,385 @@
+
+        if is_error == 1:
+            error_msg = "ERROR WHEEL TORQUE "
+            error_msg += str(is_error_tab)
+            print(self.FAIL + error_msg + self.ENDC)
+            return 0
+
+        print(self.OKGREEN + "PASSED" + self.ENDC)
+        return 1
+
+    def check_imu(self):
+        msg_cnt = 0
+        time_now = time.time()
+        imu_valid = parse_yaml(self.path + "imu.yaml")
+
+        accel_del = imu_valid["imu"]["accel_del"]
+        accel_x = imu_valid["imu"]["accel_x"]
+        accel_y = imu_valid["imu"]["accel_y"]
+        accel_z = imu_valid["imu"]["accel_z"]
+
+        gyro_del = imu_valid["imu"]["gyro_del"]
+        gyro_x = imu_valid["imu"]["gyro_x"]
+        gyro_y = imu_valid["imu"]["gyro_y"]
+        gyro_z = imu_valid["imu"]["gyro_z"]
+
+        while msg_cnt < 50:
+            if time_now + imu_valid["imu"]["timeout"] < time.time():
+                print(self.WARNING + "TIMEOUT" + self.ENDC)
+                return 0
+
+            if self.is_new_imu_data == 1:
+                time_now = time.time()
+                self.is_new_imu_data = 0
+                msg_cnt += 1
+
+                if not (
+                    accel_x - accel_del < self.imu_data.accel_x < accel_x + accel_del
+                    and accel_y - accel_del
+                    < self.imu_data.accel_y
+                    < accel_y + accel_del
+                    and accel_z - accel_del
+                    < abs(self.imu_data.accel_z)
+                    < accel_z + accel_del
+                    and gyro_x - gyro_del < self.imu_data.gyro_x < gyro_x + gyro_del
+                    and gyro_y - gyro_del < self.imu_data.gyro_y < gyro_y + gyro_del
+                    and gyro_z - gyro_del < self.imu_data.gyro_z < gyro_z + gyro_del
+                ):
+                    print(self.FAIL + "INVALID DATA" + self.ENDC)
+                    return 0
+
+        print(self.OKGREEN + "PASSED" + self.ENDC)
+        return 1
+
+    def check_battery(self):
+        msg_cnt = 0
+        time_now = time.time()
+        batt_valid = parse_yaml(self.path + "battery.yaml")
+
+        while msg_cnt < 50:
+            if time_now + batt_valid["battery"]["timeout"] < time.time():
+                print(self.WARNING + "TIMEOUT" + self.ENDC)
+                return 0
+
+            if self.is_new_battery_data == 1:
+                time_now = time.time()
+                self.is_new_battery_data = 0
+                msg_cnt += 1
+
+                if self.battery_data.data <= batt_valid["battery"]["voltage_min"]:
+                    print(self.FAIL + "LOW VOLTAGE" + self.ENDC)
+                    return 0
+                if self.battery_data.data >= batt_valid["battery"]["voltage_max"]:
+                    print(self.FAIL + "HIGH VOLTAGE" + self.ENDC)
+                    return 0
+
+        print(self.OKGREEN + "PASSED" + self.ENDC)
+        return 1
+
+    # pylint: disable=too-many-branches,too-many-statements
+    def test_hw(
```

Comment:
I'm still not convinced that the `test_hw` should be a method inside the `HardwareTester` class instead of a function outside that class that serves as an example usage of that class, but I'll leave the decision up to you.